x/shield/keeper/collateral.go

```diff
@@ -13,7 +14,8 @@ func (k Keeper) DepositCollateral(ctx sdk.Context, from sdk.AccAddress, amount s
 		provider = k.addProvider(ctx, from)
 	}
 	// Check if there are enough delegations backing collaterals.
```

Comment:
Why this is needed?

Reply:
To make sure the provider has enough delegations to deposit.